File: Practice/src/exercise/NextLargeNumber.java
package exercise;

public class NextLargeNumber {

    public int solution(int n) {
	    String s=Integer.toBinaryString(n);
	    int num=0;
	    int check=0;
	    for(int i=0;i<s.length();i++) {
	    	if(s.charAt(i)=='1') num++; 
	    }
	    while(num!=check) {
	    	check=0;
	    	n++;
	    	s=Integer.toBinaryString(n);
	    	for(int i=0;i<s.length();i++) {
		    	if(s.charAt(i)=='1') check++; 
	    	}
	    }
        return n;
    }
}

File: Practice/src/exercise/NextLargeNumber.js
function solution(n) {
  let num = 0;
  for (let i of n.toString(2)) {
    if (i == "1") num++;
  }
  while (true) {
    let check = 0;
    n++;
    for (let i of n.toString(2)) {
      if (i == "1") check++;
    }
    if (check == num) break;
  }
  return n;
}